### DISCUSSION TITLE
Can I start using RDS as my managed postgres instance? Are we there yet?

### DISCUSSION THREAD

**Discussion by clmnin** [geldata/gel#2168] [Q&A] [answered]
I tried using `edb` to setup EdgeDB in my local environment which runs postgres locally. 

It works fine - I was able to run my app successfully.

**Questions**
1. Can I use `edb` to setup EdgeDB and use a managed postgres instance like RDS? 
    * Issue edgedb#2013 mentions support for managed postgres instance using RDS (presumably because AWS let's you have access to a superuser which EdgeDB needs)
2. The CLI won't work, as mentioned by Elvis [here](https://github.com/edgedb/edgedb/issues/1825#issuecomment-742204359). But from my understanding, that is fine (for now). Or is there something the CLI offers for a (non critical / personal project) production environment?

**Comment by elprans** [comment, marked as answer]
Running on RDS is already possible via `--postgres-dsn`.  We run several EdgeDB-powered services internally that way.  The reason why we don't overtly advertise or recommend this mode is because there's no `edgedb server` CLI support for that, which, in turn, means that there's no way to upgrade and manage such instances easily yet (we are working on this: #2158 and teaching the CLI to accept `--postgres-dsn` should be the only things needed here).

You can use the development command `edb wipe --postgres-dsn=<rds-uri>` to completely remove EdgeDB data from a given cluster, which gives you the ability to install a new version without provisioning a new cluster (although simply dropping all EdgeDB databases and roles would work too).

> presumably because AWS let's you have access to a superuser which EdgeDB needs

Strictly speaking, full superuser access isn't needed, only a certain set of privileges, most notably `CREATEDB` and `CREATEROLE`.

**Reply by clmnin** [reply]
I understand the major pain point in not having the CLI is that it would be tiresome to manage an instance - backup and restore. So it makes sense to keep using the CLI and installing EdgeDB in a VM. 
But I have another question:
1. How do I restore a EdgeDB instance to RDS when I decide to move to a managed service?
    * Since the CLI doesn't support `--postgres-dsn` I can't restore my current DB to my local postgres instance. Is there a way to acheive this? 
    * I assume this is in the roadmap, and will be added to CLI in the beta phase.
2. How do I start and stop an `edb` server?
    * I tried starting a server using `edb server --postgres-dsn <postgres-url>`, but `^C` seem to be an "unsafe" way to stop the server.
    * Tried `edb server -b --postgres-dsn <postgres-url>` to daemonize it, but didn't find any processs running edgedb when I ran `ps aux | grep edgedb`

**Reply by elprans** [reply]
> How do I restore a EdgeDB instance to RDS when I decide to move to a managed service? 

Here's how:

```shell
# Dump current data
$ edgedb dump --all /backup/dir

# Bootstrap an instance on a remote cluster
$ edb server --port=<some-port> --postgres-dsn=<postgres-dsn>

# Restore into the new instance
$ edgedb --port=<some-port> restore --all /backup/dir
```

> but `^C` seem to be an "unsafe" way to stop the server.

Sending `SIGINT` or `SIGTERM` manually is the only way for now, but, once implemented, `edgedb server start|stop` would be the way.

> Tried `edb server -b --postgres-dsn <postgres-url>` to daemonize it, but didn't find any processs running

Hmm.  `--background` appears to be broken.  I recommend writing a simple systemd unit if you're on a system that runs systemd.  For example:

```
[Unit]
Description=EdgeDB server
After=syslog.target
After=network.target

[Service]
Type=notify

RuntimeDirectory=edgedb
ExecStart=<full-path-to>/edgedb-server --postgres-dsn=<postgres-dsn> --runstate-dir=%t/edgedb
ExecReload=/bin/kill -HUP ${MAINPID}
KillMode=mixed
KillSignal=SIGINT
TimeoutSec=0

[Install]
WantedBy=multi-user.target
```
